src/ec/packages/ec-component-text-input/text-input.html.twig

```diff
@@ -60,30 +66,38 @@
 <div class="ecl-form-group ecl-form-group--text-input">
   {% if _label is not empty %}
     <label
-      class="ecl-form-label{{ invalid ? ' ecl-form-label--invalid' }}{{ hide_label ? ' ecl-form-label--hidden' }}"
+      class="ecl-form-label
+      {{- _invalid ? ' ecl-form-label--invalid' -}}
+      {{- _disabled ? ' ecl-form-label--disabled' -}}
+      "
       for="{{ _id }}"
     >
       {%- block label _label -%}
+      {%- if _required %}
+        <span class="ecl-form-label__required">{{ _required_text }}</span>
+      {%- else %}
+        <span class="ecl-form-label__optional"> {{ _optional_text }}</span>
+      {%- endif -%}
     </label>
   {% endif %}
 
+  {% if _helper_text is not empty %}
+    <div class="ecl-help-block {{ _disabled ? ' ecl-help-block--disabled' }}">{% block helper_text _helper_text %}</div>
```

Comment:
this would generate a double space, right? `ecl-help-block {{ _disabled ? ' ecl`

Reply:
done.